src/target/adiv5_jtag.c

```diff
@@ -57,11 +57,36 @@ void adiv5_jtag_dp_handler(const uint8_t dev_index)
 	bmda_jtag_dp_init(dp);
 #endif
 
-	if (jtag_devs[dev_index].jd_idcode == JTAG_IDCODE_ARM_DPv0)
+	const uint32_t idcode = jtag_devs[dev_index].jd_idcode;
+
+	/* Check if we have a JTAG ID code and that it's a valid one */
+	if (idcode & 1U) {
+		/*
+		 * We do, so this is a JTAG DP. Let's pull out the designer and part codes then.
```

Comment:
In this context this part of the comment looses it's purpose, we already knew it's a JTAG DP